R/auxiliary.R

```diff
@@ -117,3 +117,20 @@ process_row_details <- function(details_json) {
 
   tmp_result
 }
+
+#' clean inputs by escaping all the special characters,
+#' to use them pattern matching
+#'
+#' @param input string that need to be cleaned
+#'
+#' @return A cleaned regex pattern after escaping special characters.
+#'
+#' @noRd
+#' @keywords internal
+clean_regex <- function(input) {
+  if (input == "\\") {
+    "\\\\"
+  } else {
+    gsub("([\\[\\]{}()*+?.^$|-])", "\\\\\\1", input, perl = TRUE)
+  }
+}
```

Comment:
Can you add some tests on this function? Maybe rename it to `escape_regex`

Alternatively, use `stringr::str_escape` and add that to `Imports`. It is installed anyway as part of our dependencies on `tidyr`

(the substitution in `stringr::str_escape` function body is a bit different)

Reply:
Good suggestion
using stringr::str_escape does the same thing as the gsub("([\\[\\]{}()*+?.^$|-])", "\\\\\\1", input, perl = TRUE)
tested it.